include/atomics/receiverCadmium.hpp

```diff
@@ -29,84 +28,88 @@ using namespace cadmium;
-              for(const auto &x : get_messages<typename defs::in>(mbs)){
-                state.ackNum = static_cast < int > (x.value);
+        // external transition
+        void external_transition(TIME e, 
+                                 typename make_message_bags<input_ports>::type mbs) { 
+            if (get_messages<typename defs::in>(mbs).size() > 1) {
+                assert(false && "one message per time uniti");    
+            } 
+            for (const auto &x : get_messages<typename defs::in>(mbs)) {
+                state.acknowledgement_number = static_cast<int>(x.value);
                 state.sending = true;
-              }  
-                           
-            }
+            }            
+        }
 
-            // confluence transition
-            void confluence_transition(TIME e, typename make_message_bags<input_ports>::type mbs) {
-                internal_transition();
-                external_transition(TIME(), std::move(mbs));
-            }
+        // confluence transition
+        void confluence_transition(TIME e, 
+                                   typename make_message_bags<input_ports>::type mbs) {
+            internal_transition();
+            external_transition(TIME(), std::move(mbs));
+        }
 
-            // output function
-            typename make_message_bags<output_ports>::type output() const {
-              typename make_message_bags<output_ports>::type bags;
-              Message_t out;              
-              out.value = state.ackNum % 10;
-              get_messages<typename defs::out>(bags).push_back(out);
-                
-              return bags;
+        // output function
+        typename make_message_bags<output_ports>::type output() const {
+            typename make_message_bags<output_ports>::type bags;
+            message_t out;              
+            out.value = state.acknowledgement_number % 10;
+            get_messages<typename defs::out>(bags).push_back(out);       
+            return bags;
+        }
 
+        // time_advance function
+        TIME time_advance() const {  
+            TIME next_internal;
+            if (state.sending) {
+                next_internal = PREPARATION_TIME;
             }
-
-            // time_advance function
-            TIME time_advance() const {  
-              TIME next_internal;
-              if (state.sending) {
-                next_internal = preparationTime;
-              }else {
+            else {
                 next_internal = std::numeric_limits<TIME>::infinity();
-              }    
-              return next_internal;
-            }
+            }    
+            return next_internal;
+        }
 
-            friend std::ostringstream& operator<<(std::ostringstream& os, const typename Receiver<TIME>::state_type& i) {
-                os << "ackNum: " << i.ackNum; 
+        friend std::ostringstream& operator<<(std::ostringstream& os, 

```

Comment:
As discussed earlier by Gagandeep, the variable names have to be self-descriptive. You can change "os" to something like "output_stream".
Thanks.

Reply:
We have changed the variable names in the file. many thanks for letting us know. 
May I request you to please review them again?